lms/static/js/instructor_dashboard/data_download.js

```diff
@@ -0,0 +1,399 @@
+                            gettext('Error generating student profile information. Please try again.')
+                        );
+                        return $('.msg-error').css({
+                            display: 'block'
+                        });
+                    },
+                    success: function(data) {
+                        dataDownloadObj.$reports_request_response.text(data.status);
+                        return $('.msg-confirm').css({
+                            display: 'block'
+                        });
+                    }
+                });
+            });
+            this.$list_studs_btn.click(function() {
+                var url;
+                url = dataDownloadObj.$list_studs_btn.data('endpoint');
+                dataDownloadObj.clear_display();
+                dataDownloadObj.$download_display_table.text(gettext('Loading'));
+                return $.ajax({
+                    type: 'POST',
+                    dataType: 'json',
+                    url: url,
+                    error: function() {
+                        dataDownloadObj.clear_display();
+                        return dataDownloadObj.$download_request_response_error.text(
+                            gettext('Error getting student list.')
+                        );
+                    },
+                    success: function(data) {
+                        var $tablePlaceholder, columns, feature, grid, gridData, options;
+                        dataDownloadObj.clear_display();
+                        options = {
+                            enableCellNavigation: true,
+                            enableColumnReorder: false,
+                            forceFitColumns: true,
+                            rowHeight: 35
+                        };
+                        columns = (function() {
+                            var i, len, ref, results;
+                            ref = data.queried_features;
+                            results = [];
+                            for (i = 0, len = ref.length; i < len; i++) {
+                                feature = ref[i];
+                                results.push({
+                                    id: feature,
+                                    field: feature,
+                                    name: data.feature_names[feature]
+                                });
+                            }
+                            return results;
+                        }());
+                        gridData = data.students;
+                        $tablePlaceholder = $('<div/>', {
+                            class: 'slickgrid'
+                        });
+                        dataDownloadObj.$download_display_table.append($tablePlaceholder);
+                        grid = new window.Slick.Grid($tablePlaceholder, gridData, columns, options);
+                        return grid;
```

Comment:
this is a case where we probably _do_ need this return but it would be cleaner I think to never declare `grid` and just `return new window.Slick.Grid`